データベース設計

kzen23 opened the pull request:
# What
furima-37082開発に必要なDB設計

# Why
furima-37082開発を進めるための指針

ER図リンク: https://gyazo.com/06124a95def042431ae2dd84d25daf02

kzen23 committed:
ER図作成

kzen23 committed:
README作成

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,65 @@
-* Database creation
+| Column              | Type       | Options                            |
+| ------------------- | -------    | ---------------------------------- |
+| name                | string     | null: false, max_length: 40        |
+| info                | text       | null: false, max_length: 1000      |
```
```
| name                | string     | null: false, max_length: 40        |
| info                | text       | null: false, max_length: 1000      |
```
上記の「max_length」もテーブル設計での記述は不要なため削除をお願いいたします！
viewでは「maxlength:"40"」という記述で文字数制限をしていますが、テーブルに制限をかけるわけではないためREADMEでの記述は不要でございます。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| name                | string     | null: false, max_length: 40        |
+| info                | text       | null: false, max_length: 1000      |
+| price               | integer    | null: false, character: half-width |
+| category            | integer    | null: false                        |
+| sales_status        | integer    | null: false                        |
+| shipping_fee_status | integer    | null: false                        |
+| prefecture          | integer    | null: false                        |
+| scheduled_delivery  | integer    | null: false                        |
```
```
| category            | integer    | null: false                        |
| sales_status        | integer    | null: false                        |
| shipping_fee_status | integer    | null: false                        |
| prefecture          | integer    | null: false                        |
| scheduled_delivery  | integer    | null: false                        |
```
プルダウンで選択する箇所の値は「id」で扱われます。
そのためカラム名を「単数形」＋「_id」に修正し、型を「数値型」に修正しましょう！

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,65 @@
+## Association
+
+belongs_to :user
+belongs_to :item
```
Addressesテーブルのアソシエーションの記述もお願いいたします。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| Column              | Type        | Options                                          |
+| ------------------- | -------     | -------------------------------------------------|
+| postal_code         | string      | null: false, character: half-width               |
+| prefecture          | integer     | null: false                                      |
```
`| prefecture | integer | null: false|`
上記もアクティブハッシュの値のため、カラム名の修正をお願いいたします。

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| city                | string      | null: false                                      |
+| addresses           | string      | null: false                                      |
+| building_name       | string      |                                                  |
+| phone_number        | integer     | null: false, character: half-width, unique: true |
```
`| phone_number | integer | null: false, character: half-width, unique: true |`
電話番号がinteger型の場合は、08012345678の先頭「0」が読み込まれないためstring型に変更しましょう！
また「unique: true」の削除をお願いいたします。

【unique: trueを削除する理由】
重複してエラーメッセージが出た場合、他人に電話番号が知られてしまうため

Saka-toku commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| addresses           | string      | null: false                                      |
+| building_name       | string      |                                                  |
+| phone_number        | integer     | null: false, character: half-width, unique: true |
+
```
ordersテーブルの外部キーカラムの記述をお願いいたします。

Saka-toku reviewed (commented):
レビュー依頼、ER図のご提出ありがとうございます！
コメントいたしましたのでご確認をお願いいたします！

kzen23 committed:
修正１

不要なmax_lengthの記述を削除。

kzen23 committed:
テーブルに記述すべきでない内容を削除

半角や全角などの指定は、モデルのファイルに記述すべき内容。

kzen23 committed:
カラム名修正

プルダウンで選択する箇所の値は、『id』で扱われるので、それに適した形に修正。

kzen23 committed:
必要なアソシエーションの追加

ordersテーブルとaddressesテーブルのアソシエーション。

kzen23 committed:
prefectureのカラム名、崩れたビューを編集。

prefectureはプルダウン形式で実装するため、カラム名を適したものにした。

kzen23 committed:
phone_numberの編集

型をstring型に変更。integer型では、090、080などの0から始まる数字に対応できないため。

unique: trueを削除。エラーが起きることで他人に電話番号の情報が知られてしまうため。

kzen23 committed:
Addressesテーブルに外部キーカラム追加

TakashiKozuma commented:
追加のレビュー依頼ありがとうございます！
修正点も含め、確認が完了しました。
LGTMです！！

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。
